### PR TITLE
Added non-serialized metrics for windows, for 3 pods

### LINKED ISSUE #104418: Add non-serialized metrics tests for Windows
#### What happened:

- Lets put a metrics CI job in place that isnt `Serial`.
- Investigate our test coverage of metrics-server
- See why our CI jobs dont test metrics-server
- Confirm that we are doing enough in `kubelet_stats`

#### What you expected to happen:

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/windows/kubelet_stats.go is a good test, lets make it part of all our containerd jobs by removing the Serial tags

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Enhancement

#### What this PR does / why we need it:
Added non-serialized metrics for windows, for 3 pods
#### Which issue(s) this PR fixes:

Fixes #104418 

#### Special notes for your reviewer:
I am not sure about this change, if this is wrong please guide me.
#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/windows/kubelet_stats.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 
 				ginkgo.By("Getting kubelet stats 5 times and checking average duration")
 				iterations := 5
-				var totalDurationMs int64
+				var totalDurationMs int
```
Comment (marosset) [inline test/e2e/windows/kubelet_stats.go:62]: duration.Milliseconds() returns int64 so we should keep this var int64 as well.
https://pkg.go.dev/time#Duration.Milliseconds

File: test/e2e/windows/kubelet_stats.go [hunk 2]
```diff
@@ -99,6 +99,70 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 		})
 	})
 })
+var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats", func() {
+	f := framework.NewDefaultFramework("kubelet-stats-test-windows")
```
Comment (marosset) [inline test/e2e/windows/kubelet_stats.go:103]: Can you make this unique (or add -serial to the one one line 39 above?)

File: test/e2e/windows/kubelet_stats.go [hunk 3]
```diff
@@ -99,6 +99,70 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
...
+		ginkgo.Context("when running 3 pods", func() {
+			// 10 seconds is the default scrape timeout for metrics-server and kube-prometheus
+			ginkgo.It("should return within 10 seconds", func() {
+
+				ginkgo.By("Selecting a Windows node")
+				targetNode, err := findWindowsNode(f)
+				framework.ExpectNoError(err, "Error finding Windows node")
+				framework.Logf("Using node: %v", targetNode.Name)
+
+				ginkgo.By("Scheduling 10 pods")
```
Comment (marosset) [inline test/e2e/windows/kubelet_stats.go:115]:
```suggestion
				ginkgo.By("Scheduling 3 pods")
```
```suggestion
				ginkgo.By("Scheduling 10 pods")
```

File: test/e2e/windows/kubelet_stats.go [hunk 4]
```diff
@@ -99,6 +99,70 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
...
+				powershellImage := imageutils.GetConfig(imageutils.BusyBox)
+				pods := newKubeletStatsTestPods(3, powershellImage, targetNode.Name)
+				f.PodClient().CreateBatch(pods)
+
+				ginkgo.By("Waiting up to 3 minutes for pods to be running")
+				timeout := 3 * time.Minute
+				e2epod.WaitForPodsRunningReady(f.ClientSet, f.Namespace.Name, 3, 0, timeout, make(map[string]string))
+
+				ginkgo.By("Getting kubelet stats 5 times and checking average duration")
```
Comment (marosset) [inline test/e2e/windows/kubelet_stats.go:124]: Is there a reason why this is querying the container stats 5 times?
I think it should be sufficient to just query stats once and ensure all of the values we expect to be non-zero are non-zero (bonus points if we can check to make sure they fall within an expected range).

File: test/e2e/windows/kubelet_stats.go (hunk 1 above)
Comment (marosset) [reply test/e2e/windows/kubelet_stats.go:62]: This should fix a lot of the build / typecheck errors too

**Comment by NikhilSharmaWe** [comment]
@marosset can you confirm on the changes made.

**Comment by NikhilSharmaWe** [comment]
@marosset this error is related to type of totalDurationMs, but I am unable to find out the cause of the error.
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105854/pull-kubernetes-verify-govet-levee/1456828807108366336

**Comment by marosset** [comment]
> @marosset this error is related to type of totalDurationMs, but I am unable to find out the cause of the error. https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/105854/pull-kubernetes-verify-govet-levee/1456828807108366336

pull-kubernetes-typecheck has some details for the error 
```
ERROR(linux/arm): /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/windows/kubelet_stats.go:132:6: invalid operation: mismatched types int and int64
ERROR(linux/arm): /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/windows/kubelet_stats.go:157:22: invalid operation: mismatched types int and int64
```

https://github.com/kubernetes/kubernetes/pull/105854/files#diff-36d32acf28aadd81b686bade1357da7735107d8037d36f1d727c2e368a82e0d7R126 likely needs to be updated to `int64`

**Comment by NikhilSharmaWe** [comment]
@marosset Is this PR ready or is there something still to be worked on.

**Comment by marosset** [comment]
/lgtm
/approve

i see the test passing in the `pull-kubernetes-e2e-aks-engine-windows-containerd` PR job.

Thanks @NikhilSharmaWe

**Comment by NikhilSharmaWe** [comment]
@marosset what should be the kind for this PR

**Comment by NikhilSharmaWe** [comment]
@marosset it is showing that the PR needs kind label, what should be the kind for this PR or can you add an appropriate one.

**Pull request merged**